tests/conftest.py

```diff
@@ -338,96 +256,95 @@ def funding_nodes(private_key):
+    with open(ANVIL_CFG_FILE, "r") as file:
+        data: dict = json.load(file)
+        private_key = data.get("private_keys", [""])[0]
 
     logging.info("Mirror contract data because of anvil-deploy node only writing to localhost")
 
-    shutil.copyfile(PROTOCOL_CONFIG_FILE, TEST_PROTOCOL_CONFIG_FILE)
-    mirror_contract_data(TEST_PROTOCOL_CONFIG_FILE, DEPLOYMENTS_SUMMARY_FILE, ANVIL_NETWORK1, ANVIL_NETWORK1)
-    mirror_contract_data(TEST_PROTOCOL_CONFIG_FILE, DEPLOYMENTS_SUMMARY_FILE, ANVIL_NETWORK1, ANVIL_NETWORK2)
+    shutil.copy(PROTOCOL_CONFIG_FILE, TEST_PROTOCOL_CONFIG_FILE)
+    mirror_contract_data(TEST_PROTOCOL_CONFIG_FILE, DEPLOYMENTS_SUMMARY_FILE, NETWORK1, NETWORK1)
+    mirror_contract_data(TEST_PROTOCOL_CONFIG_FILE, DEPLOYMENTS_SUMMARY_FILE, NETWORK1, NETWORK2)
 
-    logging.info("Reuse pre-generated identities")
-    reuse_pregenerated_identities()
-
-    nodes = NODES.copy()
+    # SETUP NODES USING STORED IDENTITIES
+    logging.info("Reuse pre-generated identities and configs")
+    copy_identities()
 
+    # CREATE LOCAL SAFES AND MODULES FOR ALL THE IDS
     logging.info("Create safe and modules for all the ids, store them in args files")
-    nodes = create_local_safes(nodes, private_key)
 
-    try:
-        for node_id, node_args in nodes.items():
-            (proc, api) = setup_node(node_args)
-            nodes[node_id]["proc"] = proc
-            nodes[node_id]["api"] = api
-
-        logging.info(f"Wait for {len(nodes)} nodes to start up")
-
-        for node_id, node_args in nodes.items():
-            while True:
-                with open(f"{node_args['dir']}.log", "r") as f:
-                    logs = f.read()
-                    if "still unfunded, " in logs or "node is funded" in logs:
-                        logging.info(f"Node {node_id} up")
-                        break
-
-                sleep(0.1)
-
-        logging.info("Funding nodes")
-        funding_nodes(private_key)
-
-        logging.info("Collect node information")
-        for node_id, node_args in nodes.items():
-            (peer_id, address) = collect_node_information(node_args)
-            nodes[node_id]["peer_id"] = peer_id
-            nodes[node_id]["address"] = address
-
-        logging.info("Node setup finished, waiting 10 seconds before proceeding")
-        sleep(10)
-
-        yield nodes
-    except Exception as e:
-        logging.error(f"Creating a 7 node cluster - FAILED: {e}")
-    finally:
-        logging.debug("Tearing down the 7 node cluster")
-        for node_id, node_args in nodes.items():
-            cleanup_node(node_args)
-        pass
+    nodes: dict[str, Node] = deepcopy(NODES)
+
+    safe_custom_env: dict = {
+        "ETHERSCAN_API_KEY": "anykey",
+        "IDENTITY_PASSWORD": PASSWORD,
+        "DEPLOYER_PRIVATE_KEY": private_key,
+        "PRIVATE_KEY": private_key,
+        "PATH": PWD.parent.joinpath(f".foundry/bin:{os.environ['PATH']}").as_posix(),
```

Comment:
This should not be needed, it is done by the direnv in `.envrc`

Reply:
Sadly without the custom_env parameters in, hopli command fails.

Reply:
Done.